cmd/vulcan-trivy/vulcan-trivy.go

```diff
@@ -103,65 +176,200 @@ func run(ctx context.Context, target, assetType, optJSON string, state checkstat
 		}
 	}
 
-	slashSplit := strings.SplitAfterN(target, "/", 2)
-	if len(slashSplit) <= 1 {
-		logger.Warnf("%s does not have a path", target)
+	trivyArgs := []string{}
+	// Skip vulnerability db update if not explicitly forced.
+	if !opt.ForceUpdateDB {
+		trivyArgs = append(trivyArgs, "--skip-update")
 	}
-	targetSplit := strings.Split(slashSplit[len(slashSplit)-1], ":")
-	if len(targetSplit) != 2 {
-		logger.Warnf("%s does not have a tag", target)
+	// Show only vulnerabilities with fixes.
+	if opt.IgnoreUnfixed {
+		trivyArgs = append(trivyArgs, "--ignore-unfixed")
 	}
 
-	registryDomain := strings.Trim(slashSplit[0], "/")
-	// If docker registry equals registryDomain, export trivy credential env vars.
-	if registryDomain == registryEnvDomain {
-		os.Setenv("TRIVY_AUTH_URL", registryEnvDomain)
-		os.Setenv("TRIVY_USERNAME", registryEnvUsername)
-		os.Setenv("TRIVY_PASSWORD", registryEnvPassword)
+	for _, p := range FilePatters {
+		trivyArgs = append(trivyArgs, []string{"--file-patterns", fmt.Sprintf(`"%s"`, p)}...)
 	}
 
-	isReachable, err := helpers.IsReachable(target, assetType,
-		helpers.NewDockerCreds(os.Getenv("TRIVY_USERNAME"), os.Getenv("TRIVY_PASSWORD")))
-	if err != nil {
-		logger.Warnf("Can not check asset reachability: %v", err)
-	}
-	if !isReachable {
-		return checkstate.ErrAssetUnreachable
+	if strings.Contains(assetType, "DockerImage") {
+		sc := checksToParam(opt.ImageChecks)
+		if sc == "" {
+			logger.Warnf("No checks enabled for DockerImage")
+			return nil
```

Comment:
In my opinion, in order to keep the same behavior as previous version/refactor, I think that if there are no `sc` configuration for DockerImages, we should scan the same we use to do: scan for vulnerabilities but no for secrets or configs.

Reply:
No strong opinion, so done.